src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/samplers/MetricsDBFileSampler.java

```diff
@@ -0,0 +1,83 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.samplers;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.AppContext;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metricsdb.MetricsDB;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.metrics.ReaderMetrics;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.stats.collectors.SampleAggregator;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.stats.emitters.ISampler;
+
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+import java.nio.file.Paths;
+import java.util.Objects;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class MetricsDBFileSampler implements ISampler {
+  private static final Logger LOG = LogManager.getLogger(MetricsDBFileSampler.class);
+  private final AppContext appContext;
+
+  public MetricsDBFileSampler(final AppContext appContext) {
+    Objects.requireNonNull(appContext);
+    this.appContext = appContext;
+  }
+
+  @Override
+  public void sample(SampleAggregator sampleCollector) {
+    String metricsdbPrefix = MetricsDB.getFilePrefix();
+
+    Path targetDirectoryPath = Paths.get(metricsdbPrefix).getParent();
+
+    String metricsdbPattern = "regex:" + metricsdbPrefix + "\\d+";
+    String metricsdbTarPattern = "regex:" + metricsdbPrefix.substring(0, metricsdbPrefix.length() - 1)
+        + ".tar.\\d+-\\d+-\\d+-\\d+-\\d+.gz";
+    PathMatcher metricsdbMatcher = FileSystems.getDefault().getPathMatcher(metricsdbPattern);
+    PathMatcher metricsdbTarMatcher = FileSystems.getDefault().getPathMatcher(metricsdbTarPattern);
+    DirectoryStream.Filter<Path> metricsdbFilter = metricsdbMatcher::matches;
+    DirectoryStream.Filter<Path> metricsdbTarFilter = metricsdbTarMatcher::matches;
```

Comment:
Certain variables between Line 32-40 can be made class static variables, move them out of the `sample()` method.

Reply:
Sure